crates/core_arch/src/macros.rs

```diff
@@ -1,8 +1,8 @@
 //! Utility macros.
 
 // Helper struct used to trigger const eval errors when a const generic immediate value is
-// out of range.
-pub(crate) struct ValidateConstImm8<const imm8: i32>();
+// out of 8-bit range.
+pub(crate) struct ValidateConstImm8<const imm8: i32>;
```

Comment:
It might be worth making this generic over the bit width:

```rust
struct ValidateConstImm8<const imm i32, const bits: i32>
```

The check would then be:
```rust
let _ = 1 / ((imm >= 0 && imm < (1 << bits)) as usize);
```

Then we can simply define macros referring to a single instance of this struct. These macros should ideally be in the generic `macros.rs` instead of the arch-specific one.

Reply:
good idea, that should be enough for all other `constify_imm`s analogues. (the other design has the advantage of encoding the immediate width in its name, and the expression shows up in the const error message, but a single struct will be better)